compiler/src/middle_end/matchcomp.re

```diff
@@ -627,6 +746,34 @@ let rec compile_matrix = mtx =>
         );
       let result = compile_matrix(mtx);
       Explode(matrix_type, result);
+    | ArrayMatrix(arity) =>
+      let arities = matrix_head_arities(mtx);
+      let handle_arity = ((_, switch_branches), arity) => {
+        let specialized = specialize_array_matrix(arity, mtx);
+        let result = compile_matrix(specialized);
+        let final_tree = Explode(ArrayMatrix(arity), result);
+        (final_tree, [(arity, final_tree), ...switch_branches]);
+      };
+
+      switch (arities) {
+      | [] =>
+        failwith(
+          "Internal error: compile_matrix: non-wildcard column returned no arities",
+        )
+      | _ =>
+        let (result, switch_branches) =
```

Comment:
Am I blind or is this `result` never used?

Reply:
It is not. I'll get that updated.